### PR TITLE
Add multi package support

### PR DESCRIPTION
**Pull request by danielkarso-pon**
Via: https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html

### CODE REVIEW

**Comment by danielkarso-pon** [comment]
PR force updated after it failed previously because of npm lint issues

**Comment by georgyangelov** [comment]
Thank you for this! Let me know after you have a chance to test the package as a dependency of some project that uses ESM.

@AGalabov could you take a look as well, also do you have somewhere we can test the CJS importing before a release?

**Review by georgyangelov — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,8 +1,15 @@
 {
   "name": "@asteasolutions/zod-to-openapi",
-  "version": "6.4.0",
+  "version": "7.0.0",
```
Comment (AGalabov) [inline package.json:3]: There is no need to modify those manually. We have an automated release process and we'll update them accordingly

File: package.json [hunk 2]
```diff
@@ -1,8 +1,15 @@
...
-  "main": "dist/index.js",
+  "main": "dist/cjs/index.js",
+  "module": "dist/mjs/index.js",
+  "exports": {
+    ".": {
+      "import": "./dist/mjs/index.js",
+      "require": "./dist/cjs/index.js"
+    }
+  },
   "types": "dist/index.d.ts",
```
Comment (AGalabov) [inline package.json:13]: Shouldn't the types be moved as well?

File: package.json (hunk 1 above)
Comment (danielkarso-pon) [reply package.json]: Fixed

File: package.json (hunk 2 above)
Comment (danielkarso-pon) [reply package.json]: Yes, that looks like something new (to me) . I followed this guide as to support nodejs and typescript for older and newer versions; https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing

**Comment by AGalabov** [comment]
@danielkarso-pon did you manage to test this? I just tried setting up some test projects. From what I see the current setup would not fully work. One thing that I saw as a problem was that the compiled js files for ESM modules are lacking the file extension `.js` at the end => the internal imports that we use in out app are "wrong" and cannot be resolved.

I think we should ultimetely try and test against the repo that was mentioned [here](https://github.com/honojs/middleware/issues/436) - that lead from hono to us

**Comment by AGalabov** [comment]
@danielkarso-pon thank you once again for the contribution, ideas and the review. I will go ahead with the other PR as we agreed

**Pull request closed without merging**